addons/mail/models/mail_activity.py

```diff
@@ -78,6 +78,7 @@ def _default_activity_type_for_model(self, model):
     automated = fields.Boolean(
         'Automated activity', readonly=True,
         help='Indicates this activity has been created automatically and not by any user.')
+    usage = fields.Char('Usage', readonly=True, help="Technical usage identifier")
```

Comment:
Would it make more sense for this to be a selection?

Reply:
Hello,

Couldn't we find a name which is a little easier to find in code? They are many unrelated occurrences.
`activity_usage` if anything.

Also why do we need an extra field rather than using `activity_type_id`?

And finally, why is the business information about a specific model stored on the activity and not on the business model? Employee/contract could have one/several fields to say which activity is the one about which expiring stuff.

Reply:
@seb-odoo Because we don't want to pollute activity types with stuff used only in some specific flows.

Also it should be stored on the activity, because the activity is linked to a specific flow, not the model. I don't know how you would know an activity is linked to an expiring work permit, without telling it on the activity itself. It is not the record itself that is the important point, but the flow during which the activity was created.

@reth-odoo Well unsure for a selection, because somehow we want to be able to add them in stable I guess. Kind of utility field that allows fine tuning activity flows (e.g. approvals, leave requests, ...). But globally both are ok to me.